src/sasjs-build/index.js

```diff
@@ -457,6 +474,190 @@ async function getDependencies(filePaths) {
+    uniqBy(
+      programsList.map((p) => p.fileRef),
+      (x) => x.toLocaleUpperCase()
+    ).length === programsList.length
+
+  if (areFileRefsUnique) {
+    return true
+  }
+
+  const duplicatePrograms = []
+  programsList.forEach((program, index, list) => {
+    const duplicates = list.filter(
+      (p, i) =>
+        i !== index &&
+        p.fileRef.toLocaleUpperCase() === program.fileRef.toLocaleUpperCase() &&
+        !duplicatePrograms.some(
+          (d) =>
+            d.fileName === p.fileName &&
+            d.fileRef.toLocaleUpperCase() === p.fileRef.toLocaleUpperCase()
+        )
+    )
+    duplicatePrograms.push(...duplicates)
+  })
+  const groupedDuplicates = groupBy(duplicatePrograms, (x) =>
+    x.fileRef.toLocaleUpperCase()
+  )
+  let errorMessage = ''
+  Object.keys(groupedDuplicates).forEach((fileRef) => {
+    errorMessage += `The following files have duplicate fileref '${fileRef}':\n${groupedDuplicates[
+      fileRef
+    ]
+      .map((d) => d.fileName)
+      .join(', ')}\n`
+  })
+  throw new Error(errorMessage)
+}
+
+export function validateFileRef(fileRef) {
+  if (!fileRef) {
+    throw new Error('Missing file ref.')
+  }
+
+  if (fileRef.length > 8) {
+    throw new Error(
+      'File ref is too long. File refs can have a maximum of 8 characters.'
+    )
+  }
+
+  if (!/^[_a-zA-Z][_a-zA-Z0-9]*/.test(fileRef)) {
+    throw new Error(
+      'Invalid file ref. File refs can only start with a letter or an underscore, and contain only letters, numbers and underscores.'
+    )
+  }
+
+  return true
+}
+
+export async function getProgramDependencies(
+  fileContent,
+  programFolders,
+  buildSourceFolder
+) {
+  programFolders = uniqBy(programFolders)
+  const programs = getProgramList(fileContent)
+  if (programs.length) {
+    const foundPrograms = []
+    await asyncForEach(programFolders, async (programFolder) => {
+      await asyncForEach(programs, async (program) => {
+        const filePath = path.join(buildSourceFolder, programFolder)
+        const filePaths = find.fileSync(program.fileName, filePath)
+        if (filePaths.length) {
+          const fileContent = await readFile(filePaths[0])
+          if (!fileContent) {
+            console.log(
+              chalk.yellowBright(`File ${program.fileName} is empty.`)
+            )
+          }
+          const programDependencyContent = getProgramDependencyText(
+            fileContent,
+            program.fileRef
+          )
+          foundPrograms.push(programDependencyContent)
+        } else {
+          console.log(
+            chalk.yellowBright(
+              `Skipping ${program.fileName} as program file was not found. Please check your SAS program dependencies.\n`
+            )
+          )
+        }
+      })
+    })
+
+    return foundPrograms.join('\n')
+  }
+
+  return ''
+}
+
+function getProgramDependencyText(fileContent, fileRef) {
+  let output = `filename ${fileRef} temp;\ndata _null;\nfile ${fileRef} lrecl=32767;\n`
```

Comment:
should be `data _null_;` (final underscore)

Reply:
Fixed.